x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorScheduling.java

```diff
@@ -92,9 +110,15 @@ public static ConnectorScheduling fromXContent(XContentParser parser) throws IOE
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
         {
-            builder.field(ACCESS_CONTROL_FIELD.getPreferredName(), accessControl);
-            builder.field(FULL_FIELD.getPreferredName(), full);
-            builder.field(INCREMENTAL_FIELD.getPreferredName(), incremental);
+            if (accessControl != null) {
+                builder.field(ACCESS_CONTROL_FIELD.getPreferredName(), accessControl);
+            }
+            if (full != null) {
+                builder.field(FULL_FIELD.getPreferredName(), full);
+            }
+            if (incremental != null) {
+                builder.field(INCREMENTAL_FIELD.getPreferredName(), incremental);
+            }
```

Comment:
Just to double-check: do you know, if Kibana and connectors framework can handle non-existing schedules correctly? I guess we'll find out during the integration anyway so probably no need to address it here.

Reply:
We set the default values for each schedule upon connector creation. With the update endpoint you can't remove the schedules, you can only update them, meaning we should never run into situation where a given schedule is missing